src/Cli/TableFormatter.php

```diff
@@ -365,4 +401,160 @@ protected function wordwrap($string, $width = 75, $break = "\n", $cut = false)
+                        array_shift($resizedWidths);
+                        if (empty($resizedWidths)) {
+                            $width = 0; // Zero it so not in sum.
+                            $width = $maxWidthWithoutBorders - array_sum($this->tableColumnWidths);
+                        }
+                    }
+
+                    if ($i === 0 || $i === $numberOfColumns - 1) {
+                        $width = $width + 2;
+                    }
+
+                    $columns[$i] = intval($width);
+                }
+            }
+        }
+
+        $horizontalBorder = $this->getTableHorizontalBorder($rows[0], $columns);
+
+        $table = $horizontalBorder . "\n";
+
+        if (! empty($headers)) {
+            $table .= $this->getTableRow($headers, $columns);
+            $table .= $horizontalBorder . "\n";
+        }
+
+        foreach ($rows as $row) {
+            $table .= $this->getTableRow($row, $columns);
+        }
+
+        $table .= $horizontalBorder;
+
+        return $table;
+    }
+
+    /**
+     * Whether the table is wrapped with borders or not.
+     *
+     * @param bool $isBorderedTable Whether the table is wrapped with borders or not.
+     */
+    public function setIsBorderedTable($isBorderedTable)
+    {
+        $this->isBorderedTable = $isBorderedTable;
+    }
+
+    /**
+     * Calculate table column widths based on the column content length.
+     *
+     * @param array<string> $row List of texts for each column.
+     */
+    protected function calculateTableColumnWidths($row)
+    {
+        foreach ($row as $i => $rowContent) {
+            $width = strlen($rowContent);
+
+            if ($width > $this->maxColumnWidth) {
+                $this->maxColumnWidth = $width;
+            }
+
+            if (! isset($this->tableColumnWidths[$i]) || $width > $this->tableColumnWidths[$i]) {
+                $this->tableColumnWidths[$i] = $width;
+            }
+        }
+    }
+
+    /**
+     * Get the table row.
+     *
+     * @param array<string> $row     List of texts for each column.
+     * @param array<int>    $columns List of maximum column widths.
+     *
+     * @return string Table row.
+     */
+    protected function getTableRow($row, $columns)
+    {
+        return trim($this->format($columns, $row)) . "\n";
+    }
+
+    /**
+     * Get the table horizontal border.
+     *
+     * @param array<string> $row     List of texts for each column.
+     * @param array<int>    $columns List of maximum column widths.
+     *
+     * @return string Table border.
+     */
+    protected function getTableHorizontalBorder($row, $columns)
+    {
+        $tableRow = $this->getTableRow($row, $columns);
+        $tableRow = explode("\n", $tableRow);
+        $firstRow = array_shift($tableRow);
+        $firstRow = trim($firstRow);
+        $border   = preg_replace('/[^|]/', '-', $firstRow);
+        $border   = preg_replace('/[|]/', '+', $border);
```

Comment:
These should rely on `$this->border` instead of hardcoding the border character.
```suggestion
		$borderChar = preg_quote($this->border, '/');
        $border     = preg_replace("/[^{$borderChar}]/", '-', $firstRow);
        $border     = preg_replace("/[{$borderChar}]/", '+', $border);
```